src/python.rs

```diff
@@ -556,17 +556,42 @@ impl<'p> Python<'p> {
         FromPyPointer::from_borrowed_ptr_or_opt(self, ptr)
     }
 
-    /// Lets the Python interpreter check for pending signals and invoke the
-    /// corresponding signal handlers. This can run arbitrary Python code.
+    /// Lets the Python interpreter check and handle any pending signals and invoke the
+    /// corresponding signal handlers.
     ///
-    /// If an exception is raised by the signal handler, or the default signal
-    /// handler raises an exception (such as `KeyboardInterrupt` for `SIGINT`),
-    /// an `Err` is returned.
+    /// Returns `Err(PyErr)` if any signal handler raises an exception.
     ///
-    /// This is a wrapper of the C function `PyErr_CheckSignals()`. It is good
-    /// practice to call this regularly in a long-running calculation since
-    /// SIGINT and other signals handled by Python code are left pending for its
-    /// entire duration.
+    /// These signals include `SIGINT` (normally raised by CTRL + C), which by default raises
+    /// `KeyboardInterrupt`. For this reason it is good practice to call this function regularly
+    /// if you have a long-running Rust function so that your users can cancel it.
```

Comment:
I tend to avoid using "you" and "we" to refer to the reader and us as implementers, and instead just describe the properties of the code:

```suggestion
    /// `KeyboardInterrupt`. For this reason it is good practice to call this function regularly
    /// as part of long-running Rust functions so that users can cancel it.
```